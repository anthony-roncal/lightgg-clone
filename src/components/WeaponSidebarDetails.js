import { React } from 'react';

const WeaponSidebarDetails = (
    {
        damageIcon,
        damageType,
        isCraftable,
        ammoType,
        seasonNum,
        powerCap,
        category,
        isInstanceItem,
        isEquippable,
        apiId
    }
) => {
    return (
        <ul className='sidebar-details'>
            <li>Deals <img src={`${damageIcon}`} alt={damageType} className='damage-type-icon' />{damageType} damage</li>
            <li>{!isCraftable && 'Not'} Craftable</li>
            <li>Uses {ammoType} ammo</li>
            <li>Introduced in Season {seasonNum}</li>
            <li>
                Current Drops
                <ul>
                    <li>
                        {powerCap === 999990 && 'No Power Cap'}
                        {powerCap !== 999990 && `Power Cap: ${powerCap}`}
                    </li>
                </ul>
            </li>
            <li>{category}</li>
            {isInstanceItem && <li>Instance Item</li>}
            <li>
                {isEquippable && `Equippable`}
                {!isEquippable && `Can't be Equipped`}
            </li>
            <li>API ID: {apiId}</li>
        </ul>
    );
}

export default WeaponSidebarDetails;