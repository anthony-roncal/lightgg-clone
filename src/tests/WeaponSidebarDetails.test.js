import React from "react";
import renderer from "react-test-renderer";
import WeaponSidebarDetails from "../components/WeaponSidebarDetails";

const mockData = {
    "defaultDamageTypeData": {
        "displayProperties": {
            "name": "Solar",
            "icon": "/common/destiny2_content/icons/DestinyDamageTypeDefinition_2a1773e10968f2d088b97c22b22bba9e.png"
        }
    },
    "equippingBlock": {
        "ammoTypeName": "Heavy"
    },
    "quality": {
        "versions": [
            {
                "powerCapData": {
                    "powerCap": 999990
                }
            }
        ],
    },
    "itemCategoryData": [
        {
            "displayProperties": {
                "name": "Power Weapon"
            }
        }
    ],
    "inventory": {
        "isInstanceItem": true,
    },
    "equippable": true,
    "hash": 1363886209
};

const BUNGIE_URL = 'https://www.bungie.net';

describe("Weapon Sidebar Details component", () => {
    it("renders correct elements and data", () => {
        const tree = renderer.create(
            <WeaponSidebarDetails
                damageIcon={`${BUNGIE_URL}${mockData.defaultDamageTypeData.displayProperties.icon}`}
                damageType={mockData.defaultDamageTypeData.displayProperties.name}
                isCraftable
                ammoType={mockData.equippingBlock.ammoTypeName}
                seasonNum={15}
                powerCap={mockData.quality.versions[0].powerCapData.powerCap}
                category={mockData.itemCategoryData[0].displayProperties.name}
                isInstanceItem={mockData.inventory.isInstanceItem}
                isEquippable={mockData.equippable}
                apiId={mockData.hash}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
