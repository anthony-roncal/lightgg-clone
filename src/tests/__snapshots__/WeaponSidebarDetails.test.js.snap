// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weapon Sidebar Details component renders correct elements and data 1`] = `
<div
  className="sidebar-details"
>
  <ul>
    <li>
      Deals 
      <img
        alt="Solar"
        className="damage-type-icon"
        src="https://www.bungie.net/common/destiny2_content/icons/DestinyDamageTypeDefinition_2a1773e10968f2d088b97c22b22bba9e.png"
      />
       
      <span
        className="damage-type-solar"
      >
        SOLAR
      </span>
       damage
    </li>
    <li>
       Craftable
    </li>
    <li>
      Uses 
      <span
        className="ammo-type-heavy"
      >
        HEAVY
      </span>
       ammo
    </li>
    <li>
      Introduced in 
      <span
        className="season"
      >
        Season 
        15
      </span>
    </li>
    <li>
      Current Drops
      <ul>
        <li>
          No Power Cap
        </li>
      </ul>
    </li>
    <li>
      Power Weapon
    </li>
    <li>
      Instance Item
    </li>
    <li>
      Equippable
    </li>
    <li>
      API ID: 
      1363886209
    </li>
  </ul>
</div>
`;
